modules/sc-mesh-secure-deployment/src/nats/comms_nats_controller.py

```diff
@@ -403,7 +419,13 @@ async def execute(self) -> None:
                         self.__status[
                             StatusType.DOWNLOAD_CERTIFICATES.value
                         ] = self.__action_certificates(resp, certificate_type.value)
-
+                        if (
+                            self.__status[StatusType.DOWNLOAD_CERTIFICATES.value]
+                            == "OK"
+                        ):
+                            # Restart CBMA with new certificates
+                            self.stop_cbma()
+                            self.setup_cbma()
```

Comment:
Maybe this line can be removed now that `self.__cbma_set_up` set to False inside the `stop_cbma` function so that the `if` below starts CBMA again :raccoon:

Reply:
My original intention was to leave setup_cbma() call to happen in that if clause later. This call is here because MDM server doesn't provide any kind of CBMA feature flag setting which causes that after connecting with MDM server we gent some configs from it that causes that  __is_cbma_feature_enabled() returns false thus CBMA would not be enabled. 

So yes, this part will be restructured in the future.